pkg/client/results/reader.go

```diff
@@ -192,10 +229,40 @@ func (t *tarFileInfo) Sys() interface{} {
 	return t.Reader
 }
 
+func cd(path string) (func(), error) {
```

Comment:
This is the secret sauce that was required to get this to work. The issue that my first version had (and probably yours) was that we specify the root directory and whenever we `Walk` the files we need the paths to all be relative to it; that messes things up since when "in" the tarball we dont have that issue.

So the easiest approach was to simply programmatically change our directory just while doing the walk functions. It doesn't mess with the shell at all, just what Go sees as the pwd.